src/ocm/components/clusterConfiguration/manifestsConfiguration/components/CustomManifestForm.tsx

```diff
@@ -65,9 +69,43 @@ export const CustomManifestForm = <CustomManifestFormValues extends FormViewMani
                 customManifest.yamlContent === manifest.manifestYaml,
             ),
         );
+        if (cluster && manifestsModified.length > 0) {
+          //See if manifests exists previously to make a patch or create a new one
+          const manifestsThatExists = manifestsModified.filter(
+            (manifest) => manifest.fakeId !== '',
+          );
+          const newManifestsToCreate = manifestsModified.filter(
+            (manifest) => manifest.fakeId === '',
+          );
+          await Promise.all(
+            manifestsThatExists.map(async (updatedManifest) => {
+              const existingFolderByFakeId = updatedManifest.fakeId?.substring(
```

Comment:
I would encapsulate obtaining `existingFolder` and `existingFileName` into a function.

```
const { folderName, fileName } = getManifestDetails(updatedManifest)
```

The implementation details at this point are irrelevant to the main process that's being done here.